gpii/node_modules/settingsHandlers/src/settingsHandlerUtilities.js

```diff
@@ -213,7 +213,10 @@ gpii.settingsHandlers.settingsToChanges = function (element) {
 };
 
 gpii.settingsHandlers.changesToSettings = function (element) {
-    return element.type === "DELETE" ? undefined : element.value;
+    return element.type === "DELETE" ? undefined : element.path ? {
```

Comment:
If we are going the route of not hacking this too badly, we should try to make this a little more generic, and copy over all properties of element *other* than value, since there is no general expectation they don't exist. A simple enough way to do this is to take a shallow copy of element using fluid.censorKeys(element, "value") https://docs.fluidproject.org/infusion/development/CoreAPI.html#fluidcensorkeystocensor-keys  - and then shallow copy it on again at the end.
But in either case we need a big comment here explaining what the special case is - that it is the SPI settings handler and particularly the high contrast use of it, together with link to the JIRA, and explain that we plan to get rid of this logic once there aren't any more examples of this in the SR